File: android/entities/src/main/java/com/tiagohs/entities/references/ReferenceMedia.kt
package com.tiagohs.entities.references

import com.google.gson.annotations.SerializedName
import com.tiagohs.entities.Quote
import com.tiagohs.entities.image.Image
import java.io.Serializable

class ReferenceMedia(
    @SerializedName("title")
    var title: String,

    @SerializedName("subtitle")
    var subtitle: String,

    @SerializedName("description")
    var description: String,

    @SerializedName("image")
    var image: Image,

    @SerializedName("mediaType")
    var mediaType: String,

    @SerializedName("link")
    var link: String,

    @SerializedName("buttonText")
    var buttonText: String? = null
): Reference(), Serializable

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/References/ReferenceMedia.swift
//
//  ReferenceMedia.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 23/02/21.
//

import Foundation
import ObjectMapper

class ReferenceMedia: Reference {
    var title: String!
    var subtitle: String!
    var description: String!
    var image: LocalImage!
    var mediaType: String!
    var link: String!
    var buttonText: String!
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        title                                                       <- map["title"]
        subtitle                                                    <- map["subtitle"]
        description                                                 <- map["description"]
        image                                                       <- map["image"]
        mediaType                                                   <- map["mediaType"]
        link                                                        <- map["link"]
        buttonText                                                  <- map["buttonText"]
    }
}
